plugins/woocommerce/tests/php/includes/class-wc-cart-test.php

```diff
@@ -96,6 +96,32 @@ public function test_add_variation_to_the_cart_invalid_variation_id() {
 		$variable_product->delete( true );
 	}
 
+	/**
+	 * @testdox variable product should not be added to the cart if variation_id=0.
+	 */
+	public function test_add_variation_to_the_cart_zero_variation_id() {
+		WC()->cart->empty_cart();
+		WC()->session->set( 'wc_notices', null );
+
+		$variable_product = WC_Helper_Product::create_variation_product();
+
+		// Add variable and variation_id=0.
+		WC()->cart->add_to_cart(
+			$variable_product->get_id(),
+			1,
+			0
+		);
+
+		// Check for cart contents.
+		$this->assertCount( 0, WC()->cart->get_cart_contents() );
```

Comment:
We shouldn't rely on other methods to evaluate the integrity of` add_to_cart()` here. If `get_cart_contents()` breaks, this test will fail.

Also, we're just checking for cart item count, if `create_variation_product()` breaks and outputs a random object with `id=0`, the test will pass even if `add_to_cart()` doesn't check for variable products anymore.

In this specific case, this should assert method return values and thrown exceptions.